### DISCUSSION TITLE
Where to get pdb for libskiasharp.dll

### DISCUSSION THREAD

**Discussion by marklam** [mono/SkiaSharp#1789] [Q&A] [answered]
(2.88.0-preview.127)

I'm trying to use an `SKRuntimeEffect` inside a `SkiaSharp.Views.Forms.SKCanvasView` under Windows on x64, and I'm getting an "Illegal Instruction" native exception.

The same effect works in an Avalonia app (painted via `Avalonia.Rendering.SceneGraph.ICustomDrawOperation`) so I hoped that the `.pdb` for `libSkiaSharp.dll` might help give some insight into what's going wrong.

Could anyone point me towards a pdb for the dll, as distributed in https://www.nuget.org/packages/SkiaSharp.NativeAssets.Win32/2.88.0-preview.127

BTW: I've tried reducing the effect to one that just returns black for every pixel - that still crashes, and I've tried painting a shader for `SKShader.CreateLinearGradient` - that doesn't crash; so it's something to do with using `SKRuntimeEffect`.

**Comment by mattleibow** [comment, marked as answer]
You can get all the build artifacts from https://aka.ms/skiasharp-eap/index.json

I need to put a table together, but the `2.88.0-preview.127` release was commit d5187387149d680645e1dbefbb726c50bfab46e9 This means you can download the `_NativeAssets.*` packages from the feed and extract the binaries you need. The pattern for the version is `0.0.0-commit.d5187387149d680645e1dbefbb726c50bfab46e9.*`.


So, for that specific commit, you can download the `_NativeAssets.windows` version `0.0.0-commit.d5187387149d680645e1dbefbb726c50bfab46e9.127`:

https://dev.azure.com/xamarin/public/_packaging?_a=package&feed=SkiaSharp&package=_NativeAssets.windows&protocolType=NuGet&version=0.0.0-commit.d5187387149d680645e1dbefbb726c50bfab46e9.127&view=overview

You can look at all the `_NativeAssets` and test with any commit as well as any OS - eg ios/android/windows

**Reply by mattleibow** [reply]
I do also have a PR that will push the symbols to nuget, but that is just for Windows: https://github.com/mono/SkiaSharp/pull/1797

However, the packages on the preview feed will always have all the symbols from the native builds.

**Reply by marklam** [reply]
Thanks, I can now see that the exception being thrown was
```c
    ABORT("ByteCode interpreter not enabled");
```
